languages/javascript/react-testing.md

```diff
@@ -2,95 +2,16 @@
 
 ## Overview
 
-The React testing ecosystem is constantly evolving, and like many areas of the Node world, it is made up of many small tools rather than an all-encompassing test-running solution. This guide is intended to describe what decisions we've made for unit-testing our React apps, and is not intended to be a thorough how-to guide on setting up an app for testing. It should be seen solely as a snapshot of what we're doing as of mid 2016.
+We're old hands at React testing, having lived through the eras of Jasmine, Mocha, Chai, Sinon, Enzyme, et cetera. But testing has evolved mostly thanks to Jest and React Testing Library covering all of these cases of these older, smaller tools, and so have we!
 
-Since we like to live on the cutting edge, our Node apps make use of ES2015/2016 features which are then transpiled using Babel. This complicates certain factors in regards to testing that we'll try to cover.
+## Unit and integration testing: Jest
 
-## Unit testing: Mocha
+We use [Jest](https://jestjs.io/) as our testing framework. We make use of Jest's assertions, matchers, mocks, built-in code coverage, and more.
 
-We use [Mocha](https://mochajs.org/) as our unit-testing framework. It provides us with an easy command-line interface, hooks, and dead-simple callbacks for asynchronous tests.
+We use Jest to write unit tests for code outside of React components, such as API routes and helpers. We also use Jest to write integration tests with the help of React Testing Library.
 
-In order to test transpiled files, we need to pass [`babel-register`](https://babeljs.io/docs/usage/require/) as a compiler, like so:
+We often strive toward 100% code coverage when reasonable. If necessary due to JSDOM limitations, third-party code which is hard to partially mock, or Jest's inability to pass tests with uncaught errors, we tell Istanbul, Jest's code coverage tool, to `istanbul ignore` particular lines, followed by a descriptive comment explaining why.
```

Comment:
This sentence here is pretty long and hard to parse.  Might be better split up into multiple sentences. 
I also like how the previous version discussed code-coverage in its own section.